core/templates/dev/head/pages/exploration_editor/statistics_tab/LearnerActionRenderService.js

```diff
@@ -0,0 +1,162 @@
+        stateName, timeSpentInStateSecs) {
+      var htmlString =
+        '<span class="learner-action">Left the exploration after spending a ' +
+        'total of ' + timeSpentInStateSecs + ' seconds on card "' + stateName +
+        '".</span>';
+      return htmlString;
+    };
+
+    var renderContinueButtonSubmitActionHTML = function(
+        stateName, timeSpentInStateSecs) {
+      var htmlString =
+        '<span class="learner-action">Pressed "Continue" to move to card "' +
+        stateName + '" after ' + timeSpentInStateSecs + ' seconds.</span>';
+      return htmlString;
+    };
+
+    var renderAnswerSubmitActionHTML = function(
+        answer, destStateName, timeSpentInStateSecs, currentStateName) {
+      var htmlString;
+      if (currentStateName === destStateName) {
+        htmlString =
+          '<span class="learner-action">Submitted answer "' + answer +
+          '" in card "' + currentStateName + '".</span>';
+      } else {
+        htmlString =
+          '<span class="learner-action">Submitted answer "' + answer +
+          '" and moved to card "' + destStateName + '" after spending ' +
+          timeSpentInStateSecs + ' seconds on card "' + currentStateName +
+          '".</span>';
+      }
+      return htmlString;
+    };
+
+    var renderLearnerActionsTableForMultipleIncorrectIssue = function(
+        finalBlock) {
+      var index = finalBlock.length - 1;
+      var stateName =
+        finalBlock[index].actionCustomizationArgs.state_name.value;
+
+      var tableHTML =
+        '<table class="learner-actions-table"><tr><th>Answer</th>' +
+        '<th>Feedback</th></tr>';
+      for (var i = 0; i < index; i++) {
+        if (finalBlock[i].actionType !== ACTION_TYPE_ANSWER_SUBMIT) {
+          continue;
+        }
+        var answer = finalBlock[i].actionCustomizationArgs.answer.value;
+        var feedback = finalBlock[i].actionCustomizationArgs.feedback.value;
+        tableHTML +=
+          '<tr><td>' + answer + '</td><td>' + feedback + '</td></tr>';
+      }
+      tableHTML += '</table>';
+      return tableHTML;
+    };
+
+    var renderLearnerActionHTML = function(learnerAction) {
+      var actionType = learnerAction.actionType;
+      var custArgs = learnerAction.actionCustomizationArgs;
+      if (actionType === ACTION_TYPE_EXPLORATION_START) {
+        return renderExplorationStartActionHTML(custArgs.state_name.value);
+      } else if (actionType === ACTION_TYPE_EXPLORATION_QUIT) {
+        return renderExplorationQuitActionHTML(
+          custArgs.state_name.value, custArgs.time_spent_in_state_secs.value);
+      } else if (actionType === ACTION_TYPE_ANSWER_SUBMIT) {
+        interactionId =
+          custArgs.interaction_id.value;
+        if (interactionId === 'Continue') {
+          return renderContinueButtonSubmitActionHTML(
+            custArgs.dest_state_name.value,
+            custArgs.time_spent_in_state_secs.value);
+        }
+        if (custArgs.answer.value) {
```

Comment:
Should this be an else-if? If not, please add a comment explaining why.

Reply:
Actually yes. Thanks for pointing out.